File: android/src/main/java/expo/module/signature/models/KeySpec.kt
package expo.module.signature.models

import expo.modules.kotlin.records.Field
import expo.modules.kotlin.records.Record

data class KeySpec(
    @Field val algorithm: SignatureAlgorithm = SignatureAlgorithm.EC,
    @Field val alias: String = "default_expo_signature_alias",
    @Field val size: Int = 256,
): Record

File: ios/models/KeySpec.swift
//
//  KeySpec.swift
//  SignatureModule
//
//  Created by Riccardo Pizzoni on 14/07/24.
//

import ExpoModulesCore

struct KeySpec: Record {
    @Field var algorithm: SignatureAlgorithm = .EC
    @Field var alias: String = "default_expo_signature_alias"
    @Field var size: Int = 256
}

extension KeySpec {
    var tag: Data {
        self.alias.data(using: .utf8)!
    }
}
